lib/_tls_wrap.js

```diff
@@ -426,7 +426,11 @@ TLSSocket.prototype._init = function(socket, wrap) {
 
     // Destroy socket if error happened before handshake's finish
     if (!self._secureEstablished) {
-      self.destroy(self._tlsError(err));
+      if (!self._controlReleased) {
+        self.destroy(err);
+      } else {
+        self.destroy(self._tlsError(err));
+      }
```

Comment:
Furthermore, `_tlsError` returns `null` if `self._controlReleased === false`. I don't see how this particular change makes it behave differently.